gax-httpjson/src/main/java/com/google/api/gax/httpjson/ProtoRestSerializer.java

```diff
@@ -51,6 +58,29 @@
 public class ProtoRestSerializer<RequestT extends Message> {
   private final TypeRegistry registry;
 
+  // well-known types obtained from
+  // https://github.com/googleapis/gapic-showcase/blob/fe414784c18878d704b884348d84c68fd6b87466/util/genrest/resttools/populatefield.go#L27
```

Comment:
I proposed the list from gapic-showcase but I don't think we should just use it, showcase is only used for testing so the list may get out of date. We should probably go through the protobuf doc and add any type that have special handling to the list. For now, since Go already implemented it, we can probably follow what they did [here](https://github.com/googleapis/gapic-generator-go/blob/54f801741cd6c32e67dde2e240c7b0b7af7bd2ee/internal/gengapic/genrest.go#L56).

Reply:
For sure, I can update the list of well-known types following the [docs list](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#value), but I'm curious if there is a better way to programatically obtain a list of well-known types so we can update it as a dependency or so.

We have [this documentation entry](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#value) as a source of well-known types but we may have to manually check from time to time to keep our hardcoded list up to date.
However, I noted that the official definitions such as [duration.proto](https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/duration.proto) would specify a special dotnet package `Google.Protobuf.WellKnownTypes`, whereas in java it falls into the common `com.google.protobuf` package, leaving us with no reliable programatic way to obtain a list of well-known types that would get updated as a dependency. As a last (and bad) source, there is a unit test in the same folder of the duration definition that contains a list of well-known types, but it is a [unit test](https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/unittest_well_known_types.proto), so I don't think we have too many options